src/makeInfinityAdder.js

```diff
@@ -3,8 +3,24 @@
 /***
  * @return {function}
  */
-function makeInfinityAdder() {
-  // write code here
+function makeInfinityAdder(a) {
+  let sum = 0;
+
+  function inner(b) {
```

Comment:
The parameter 'b' should not have a default value or check for `undefined` according to the additional rules. The check for `undefined` should be removed as it's not needed for the implementation.